WalletWasabi.Gui/Program.cs

```diff
@@ -40,10 +36,22 @@ private static async Task Main(string[] args)
 				TaskScheduler.UnobservedTaskException += TaskScheduler_UnobservedTaskException;
 
 				runGui = await CommandInterpreter.ExecuteCommandsAsync(Global, args);
+				
 				if (!runGui)
 				{
 					return;
 				}
+
+				if (Global.InitializeUiConfigAsync().Result)
```

Comment:
await

Reply:
must not call await in Main on OSX! if the code you await hits and `await` statement, this will change the thread ID of the UIThread on OSX, and cocoa will crash!

You only get away with it when calling `runGui = await CommandInterpreter.ExecuteCommandsAsync(Global, args);` because when the UI is going to run, it returns immediately without doing an actual await.

Reply:
In fact due to this, maybe we should remove async from Main all together and make the CommandInterpreter synchronous? If I leave a comment, then it should prevent someone breaking OSX in future by making an await inside Main?

Reply:
Umm I didn't know that. In that case it makes sense. However, that would be a different PR because it is not directly related and because it needs careful testing.

Reply:
Sure

Reply:
Can you point me to something about it? If this is indeed the case, then we must make all our `Main` methods synchronous to ensure cross platform compatibility.

Reply:
Well technically you could implement your own SynchronizationContext something like:
https://gist.github.com/kekekeks/c244a278cf73c9af444d3d7397bf858f

and that would work around it, but I wouldnt be comfortable doing this!

Reply:
Cocoa apis do a thread check to see if you are the UIThread and crash the app if you not, and the check that they make is basically a call to https://www.freebsd.org/cgi/man.cgi?query=pthread_main_np&sektion=3&apropos=0&manpath=FreeBSD+7.1-RELEASE

(First thread of the program)

Reply:
BTW, this only applies to Avalonia / UI applications, console only apps it doesnt matter.

Reply:
Same here about `.GetAwaiter().GetResult()`